### PR TITLE
Added "Managing Container Images" section (DockerHub rate limiting)

### PR DESCRIPTION
**Pull request by alexmilowski**
Added section about managing container image source with a notes about dockerhub rate limiting

### CODE REVIEW

File: content/platforms/kubernetes/managing-images.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+If you require pull secrets, you must add `imagePullSecrets`:
+
+```YAML
+apiVersion: app.redislabs.com/v1
+kind: RedisEnterpriseCluster
+metadata:
+  name: rec
+spec:
+  nodes: 3
+  imagePullSecrets:
```
Comment (amiramm) [inline content/platforms/kubernetes/managing-images.md:281]:
```suggestion
  pullSecrets:
    -name: regcred
```
@alexmilowski this is the correct notation within REC

File: content/platforms/kubernetes/managing-images.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+  - name: regcred
```
Comment (amiramm) [inline content/platforms/kubernetes/managing-images.md:282]:
```suggestion
```

**Review by amiramm — CHANGES_REQUESTED** [review]
Very well written - one minor fix. Thanks @alexmilowski !

**Review by amiramm — APPROVED** [review]
LGTM

File: content/platforms/kubernetes/managing-images.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+different deployments to consider:
+
+ 1. The Redis Enterprise operator.
+ 2. The Redis Enterprise pods and Service Rigger created by the operator.
+
+For (1), the operator container image is controlled directly by the deployment
+bundle. Whereas, for (2), the Redis Enterprise cluster pod (RS and bootstrapper) and Service Rigger
+images are controlled in the Redis Enterprise Custom Resource.
+
+Also, depending on your Kubernetes platform, your private repository may
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:159]: Again, I suggest choosing "registry" or "repository" but not both.

File: content/platforms/kubernetes/managing-images.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+require authentication. If you are using a private repository associated with
+your K8s cluster, you may be able to pull the images without credentials. For
+example, a GKE cluster in the same project as the GCR registry will have the
+necessary authorization to pull the images. Otherwise, you may
+need to add a [pull secret](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/) to your namespace
+and then tell Kubernetes and the operator to use the pull secret.
+
+If you need to specify credentials, each of the following sections will
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:167]:
```suggestion
See the following two sections for how to specify registry credentials.
```

File: content/platforms/kubernetes/managing-images.md [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+detail where the pull secret needs to be specified.
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:168]:
```suggestion

```

File: content/platforms/kubernetes/managing-images.md [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+
+### Specifying the operator image
+
+The operator bundle (e.g., see the [6.0.8-1 bundle.yaml](https://raw.githubusercontent.com/RedisLabs/redis-enterprise-k8s-docs/6.0.8-1/bundle.yaml))
+contains the operator deployment and the reference to the operator image (redislabs/operator). To use a private repository, you must
+change this image reference before you deploy the operator. This image
+should point to the same value tag pushed to the private repository:
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:175]:
```suggestion
should point to the same tag pushed to the private repository:
```

File: content/platforms/kubernetes/managing-images.md [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+    spec:
+      serviceAccountName: redis-enterprise-operator
+      containers:
+        - name: redis-enterprise-operator
+          image: gcr.io/yourproject/redislabs/operator:6.0.8-1
+...
+```
+
+Note that if you apply this change to modify an existing operator deployment,
+the operator's pod will restart. As the operator is stateless, it will pull
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:206]:
```suggestion
the operator's pod will restart. The operator, because it is stateless, will pull
```

File: content/platforms/kubernetes/managing-images.md [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+the image from the private repository (as necessary) and restart the pod. This
+kind of change will not affect any existing cluster managed by the operator.
+
+If your registry requires a pull secret, the standard `imagePullSecrets`
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:210]:
```suggestion
If your registry requires a pull secret, you may specify the standard `imagePullSecrets`
```

File: content/platforms/kubernetes/managing-images.md [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+may be specified on the operator deployment:
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:211]:
```suggestion
 on the operator deployment:
```

File: content/platforms/kubernetes/managing-images.md [hunk 10]
```diff
@@ -0,0 +1,295 @@
...
+      imagePullSecrets:
+      - name: regcred
+```
+
+### Specifying the Redis Enterprise cluster images
+
+A Redis Enterprise cluster managed by the operator consists of three
+container images:
+
+ * **`redislabs/redis`** - the Redis Enterprise container image
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:227]:
```suggestion
 * **`redislabs/redis`**: the Redis Enterprise container image
```

File: content/platforms/kubernetes/managing-images.md [hunk 11]
```diff
@@ -0,0 +1,295 @@
...
+ * **`redislabs/operator`** - the bootstrapper container image packaged with the operator
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:228]:
```suggestion
 * **`redislabs/operator`**: the bootstrapper container image packaged with the operator
```

File: content/platforms/kubernetes/managing-images.md [hunk 12]
```diff
@@ -0,0 +1,295 @@
...
+ * **`redislabs/k8s-controller`** - the service rigger container image
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:229]:
```suggestion
 * **`redislabs/k8s-controller`**: the Service Rigger container image
```

Comment (banker) [reply content/platforms/kubernetes/managing-images.md:229]: Use colons, not hyphens, to separate definitions in lists

File: content/platforms/kubernetes/managing-images.md [hunk 13]
```diff
@@ -0,0 +1,295 @@
...
+
+By default, when a new cluster is created by the operator, it uses the
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:231]:
```suggestion
By default, a new cluster created by the operator will use the
```

File: content/platforms/kubernetes/managing-images.md [hunk 14]
```diff
@@ -0,0 +1,295 @@
...
+standard container images listed above. As such, it will not necessarily pull
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:232]:
```suggestion
standard container images listed above. The operator will not necessarily pull
```

Comment (banker) [reply content/platforms/kubernetes/managing-images.md:232]: Not sure what the antecedent of "it" is. Is it the operator or the cluster?

File: content/platforms/kubernetes/managing-images.md [hunk 15]
```diff
@@ -0,0 +1,295 @@
...
+them from the same private repository from which the operator may have been
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:233]:
```suggestion
the images from the same private repository that the operator came from.
```

File: content/platforms/kubernetes/managing-images.md [hunk 16]
```diff
@@ -0,0 +1,295 @@
...
+pulled. The images will be pulled from the default container registry for
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:234]:
```suggestion
The images will be pulled from the default container registry for
```

File: content/platforms/kubernetes/managing-images.md [hunk 17]
```diff
@@ -0,0 +1,295 @@
...
+the Kubernetes cluster (i.e., typically DockerHub).
+
+The consequence is that you must specify all three container images
+to pull all the images from the same private repository. The container
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:238]: Can you rephrase the first sentence and be more specific?

File: content/platforms/kubernetes/managing-images.md [hunk 18]
```diff
@@ -0,0 +1,295 @@
...
+images are controlled by an image specification that specifies the
+image pull policy, the container image, and the version tag.
+
+The corresponding image spec labels are:
+
+ * **`redisEnterpriseImageSpec`** - controls the Redis Enterprise container image, *the version should match the RS version associated with the operator version*
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:244]: Replace hyphens with colons here

Comment (banker) [inline content/platforms/kubernetes/managing-images.md:244]:
```suggestion
 * **`redisEnterpriseImageSpec`**: controls the Redis Enterprise container image. *The version should match the RS version associated with the operator version*.
```

File: content/platforms/kubernetes/managing-images.md [hunk 19]
```diff
@@ -0,0 +1,295 @@
...
+ * **`bootstrapperImageSpec`** - controls the bootstrapper container image, *the version must match the operator version*
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:245]:
```suggestion
 * **`bootstrapperImageSpec`**": controls the bootstrapper container image. *The version must match the operator version*.
```

File: content/platforms/kubernetes/managing-images.md [hunk 20]
```diff
@@ -0,0 +1,295 @@
...
+ * **`redisEnterpriseServicesRiggerImageSpec`** - controls the service rigger version, *the version must match the operator version*
```
Comment (banker) [inline content/platforms/kubernetes/managing-images.md:246]:
```suggestion
 * **`redisEnterpriseServicesRiggerImageSpec`**: controls the service rigger version. *The version must match the operator version*.
```

**Review by banker — CHANGES_REQUESTED** [review]

File: content/platforms/kubernetes/managing-images.md [hunk 21]
```diff
@@ -0,0 +1,295 @@
+---
+Title: Managing Container Images
+description: The operator and Redis Enterprise images can be configured to be pulled from a variety of sources. This section describes how to configure alternate private repositories for images or techniques for handling public repositories and possible rate limiting.
+weight: 92
+alwaysopen: false
+categories: ["Platforms"]
+aliases:
+---
+
+Redis Enterprise, the Redis Enterprise operator, and the Service Rigger
```
[the remark this answers is not in the document]
Comment (alexmilowski) [reply content/platforms/kubernetes/managing-images.md]: There is no definition of the Service Rigger - which is gross oversight in the docs.

[the remark this answers is not in the document]
Comment (alexmilowski) [reply content/platforms/kubernetes/managing-images.md]: I will have to write an architecture overview PR that addresses this.

File: content/platforms/kubernetes/managing-images.md [hunk 22]
```diff
@@ -0,0 +1,295 @@
...
+docker pull redislabs/k8s-controller:${OPERATOR_VERSION}
+docker tag redislabs/redis:${RS_VERSION} ${PRIVATE_REPO}/redislabs/redis:${RS_VERSION}
+docker tag redislabs/operator:${OPERATOR_VERSION} ${PRIVATE_REPO}/redislabs/operator:${OPERATOR_VERSION}
+docker tag redislabs/k8s-controller:${OPERATOR_VERSION} ${PRIVATE_REPO}/redislabs/k8s-controller:${OPERATOR_VERSION}
+docker push ${PRIVATE_REPO}/redislabs/redis:${RS_VERSION}
+docker push ${PRIVATE_REPO}/redislabs/operator:${OPERATOR_VERSION}
+docker push ${PRIVATE_REPO}/redislabs/k8s-controller:${OPERATOR_VERSION}
+```
+
+## Using a private repository
```
[the remark this answers is not in the document]
Comment (alexmilowski) [reply content/platforms/kubernetes/managing-images.md:145]: It isn't necessary Docker ... "container registry" is the right term

File: content/platforms/kubernetes/managing-images.md (hunk 13 above)
Comment (alexmilowski) [reply content/platforms/kubernetes/managing-images.md:229]: I rewrote this paragraph.

File: content/platforms/kubernetes/managing-images.md (hunk 22 above)
[the remark this answers is not in the document]
Comment (alexmilowski) [reply content/platforms/kubernetes/managing-images.md]: I fixed this in the latest commit.

File: content/platforms/kubernetes/managing-images.md (hunk 3 above)
Comment (alexmilowski) [reply content/platforms/kubernetes/managing-images.md]: clarified use of container registry vs repository throughout the doc

File: content/platforms/kubernetes/managing-images.md (hunk 17 above)
Comment (alexmilowski) [reply content/platforms/kubernetes/managing-images.md]: I reworked this whole paragraph and the preceding paragraph.

File: content/platforms/kubernetes/managing-images.md [hunk 23]
```diff
@@ -0,0 +1,295 @@
...
+is to push the required images to a private repository under your control.
+
+## Managing image sources
+
+Redis Enterprise operator and cluster images are pushed to DockerHub,
+Red Hat, and other public registries for distribution. Your organization may
+require these images be copied to other registries used by your Kubernetes
+clusters.
+
+### Creating a private repository
```
[the remark this answers is not in the document]
Comment (alexmilowski) [reply content/platforms/kubernetes/managing-images.md]: clarified use of container registry vs repository throughout the doc

File: content/platforms/kubernetes/managing-images.md [hunk 24]
```diff
@@ -0,0 +1,295 @@
...
+
+ * Redis Enterprise
+ * bootstrapping a Redis Enterprise cluster node
+ * the Service Rigger
+ * the Redis Enterprise operator
+
+Once you have setup a private repository, you will need to push these images
+to your private repository. The process to do so is:
+
+ 1. Pull the various images locally for Redis Enterprise and the operator.
```
[the remark this answers is not in the document]
Comment (alexmilowski) [reply content/platforms/kubernetes/managing-images.md:124]: Various - because there can be more than one. Also, because it depends on which images you are pull and pushing and for what purpose. This is a general procedure.

File: content/platforms/kubernetes/managing-images.md [hunk 25]
```diff
@@ -0,0 +1,295 @@
...
+## Using a private repository
+
+After pushing your desired version to your private repository, you just need to
+configure your deployments to use the private repository. There are two
+different deployments to consider:
+
+ 1. The Redis Enterprise operator.
+ 2. The Redis Enterprise pods and Service Rigger created by the operator.
+
+For (1), the operator container image is controlled directly by the deployment
```
[the remark this answers is not in the document]
Comment (alexmilowski) [reply content/platforms/kubernetes/managing-images.md]: configured

File: content/platforms/kubernetes/managing-images.md [hunk 26]
```diff
@@ -0,0 +1,295 @@
...
+bundle. Whereas, for (2), the Redis Enterprise cluster pod (RS and bootstrapper) and Service Rigger
+images are controlled in the Redis Enterprise Custom Resource.
```
[the remark this answers is not in the document]
Comment (alexmilowski) [reply content/platforms/kubernetes/managing-images.md]: same = configured

File: content/platforms/kubernetes/managing-images.md [hunk 27]
```diff
@@ -0,0 +1,295 @@
...
+ * Redis Enterprise
+ * bootstrapping a Redis Enterprise cluster node
+ * the Service Rigger
+ * the Redis Enterprise operator
+
+Once you have setup a private repository, you will need to push these images
+to your private repository. The process to do so is:
+
+ 1. Pull the various images locally for Redis Enterprise and the operator.
+ 2. Tag the local images with the repository prefixed onto the name.
```
[the remark this answers is not in the document]
Comment (alexmilowski) [reply content/platforms/kubernetes/managing-images.md]: private container registry ... clarified in commit.

**Review by banker — APPROVED** [review]
